packages/web-frontend/src/sagas.js

```diff
@@ -594,7 +594,22 @@ function* fetchOrgUnitData(organisationUnitCode, projectCode) {
       includeCountryData: organisationUnitCode !== projectCode, // We should pull in all country data if we are within a project
     };
     const requestResourceUrl = `organisationUnit?${queryString.stringify(urlParameters)}`;
-    const orgUnitData = yield call(request, requestResourceUrl);
+
+    let orgUnitData = yield call(request, requestResourceUrl);
+
+    const state = yield select();
+
+    // If it's a project, use the map bounds from the store that we already have as they are more precise
+    if (orgUnitData.type === 'Project' && state.map.position.bounds) {
```

Comment:
This feels pretty fragile to me. We're relying on the fact that the `state.map.position.bounds` has been set by the project, which assumes a specific order of events. 

If it were pulling from somewhere that wasn't writeable by things other than the project (e.g. `state.something.projectx.bounds`) I'd be more keen on it.

Would still prefer it came from the backend endpoint handler though - was that a no go?

Reply:
Yes agreed. Have moved the logic to the endpoint handler